File: Number_Guessing_Game.c
// Copyright (c) 2022 Nina Mathew All rights reserved
//
// Created by: Nina Mathew
// Created on: Mar 29, 2023
// This program guesses a number

#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <time.h>


int main() {
// this function allows the user to guess a number
// and the program decides if the user is correct


    unsigned int seed = time(NULL);
    int randomNumber = rand_r(&seed) % 10;
    int NumberGuessed = 0;


    // input

    int total_Guesses = 0;
    int scanfErrorCode = 0;


    // process & output
    while (1) {
    printf(" Guess a number between 0 and 10 : ");
    scanfErrorCode = scanf("%d", &NumberGuessed);

    if (scanfErrorCode == 1 && NumberGuessed >= 0 && NumberGuessed <= 10) {
        if (NumberGuessed > randomNumber) {
            printf("Too high, try again \n");
        } else if (NumberGuessed < randomNumber) {
            printf("Too low, try again \n");
        } else {
            printf("Correct 😊 \n");
            break;
        }
    } else {
        printf("Invalid input, try again \n");
            break;
        }
    }
    printf("\nDone.\n");
    return 0;
}

File: number_guessing_game.py
#!/usr/bin/env python3
"""
Created by: Nina Mathew
Created on: March 21 2023
This program is the number guessing game
"""


import random


def main() -> None:
    random_number = random.randint(1, 10)
    total_guesses = 0
    number_guessed_string = input("Guess the number between 0 and 10 : ")
    try:
        number_guessed = int(number_guessed_string)
        for total_guesses in range(5):
            number_guessed_string = input("Guess the number between 0 and 10 : ")
            if number_guessed < random_number:
                print("Your guess is below the number, please guess again")
            elif number_guessed > random_number:
                print("Your guess is above the number, please guess again")
            else:
                break

        if number_guessed == random_number:
            total_guesses = str(total_guesses + 1)
            print("CORRECT 😊")
            if number_guessed != random_number:
                random_number = str(random_number)
                print("Incorrect 😔, Please Try Again")
    except ValueError:
        print("Invalid input, try again.")


if __name__ == "__main__":
    main()
